content/python/concepts/functions/terms/arguments-parameters/arguments-parameters.md

```diff
@@ -0,0 +1,154 @@
+
+## Parameters
+
+Parameters are variables that are declared in the function definition. They are usually processed in the function body to produce the desired result. When the function is called, each parameter is assigned the value which was passed as a corresponding argument.
+
+For example, the function below contains parameters for a `character`, a `setting`, and a `skill`, which are used as inputs to write the first sentence of a book.
+
+```py
+def write_a_book(character, setting, skill):
+  print(character + " is in " +
+        setting + " practicing " +
+        skill + ".")
+
+write_a_book("Naomi", "engineering school", "welding")
+```
+
+The output will look like this:
+
+```shell
+Naomi is in engineering school practicing welding.
+```
+
+## Default Parameter Values
+
+Function parameters can also be initialized to a default value. In the `calc_total()` function, there are `amount` and `discount` parameters.
+
+- When the `discount` value is explicitly specified in the function call, that value is used.
+- Otherwise, the default value of 10 is used.
+
+```codebyte/python
+def calc_total(amount, discount=10):
+  total = amount * (1 - 0.01 * discount)
+  return total
+
+calc_total(100)     # Output: 90.0
+calc_total(250, 5)  # Output: 237.5
+```
+
+## Arguments
+
+Unless otherwise specified, arguments passed into a called function are assigned to each parameter in the order in which they appear in the function definition. Thus, they are also known as "positional arguments".
+
+### Keyword Arguments
+
+Python also supports keyword arguments — prefixing arguments with the names of parameters to assign them directly, regardless of the order.
+
+```py
+def write_a_book(name, color, clothing_item):
+  print(name + " was wearing a " + color +
+        " " + clothing_item + ".")
+
+write_a_book(color="yellow", clothing_item="raincoat", name="Jonas")
+```
+
+The output will look like this:
+
+```shell
+Jonas was wearing a yellow raincoat.
+```
+
+Keyword arguments must be passed after positional arguments.
+
+```py
+write_a_book(name="Jonas", "yellow", "raincoat")
+```
+
+The call above raises the following exception:
+
+```shell
+SyntaxError: positional argument follows keyword argument
+```
+
+### Varying Arguments
+
+When defining a function, it may not be necessary to know in advance how many arguments will be needed. In such cases, a special parameter `*args` is passed in. The asterisk, known in this context as the "packing operator", packs the arguments into a [tuple](https://www.codecademy.com/resources/docs/python/tuples) stored in `args`. This tuple can then be iterated through within the function.
+
+In the example below, the `multiply()` function returns the product of all numbers used in the function call.
+
+```py
+def multiply(*args):
+  product = 1
+  for arg in args:
+    product *= arg
+  return product
+
+multiply(21, 24)
```

Comment:
Should lines 103 & 104 be printed as well?

Reply:
Indeed! 😄